web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -205,8 +220,66 @@ export const BuyCreditsForm = ({
     ],
   );
 
+  const goToPaymentInfo = useCallback(
+    () => handleActiveStep(1),
+    [handleActiveStep],
+  );
+
+  const allowedDenoms = allowedDenomsData?.allowedDenoms;
+
+  const currency = data?.[CURRENCY];
+  const creditsAmount = data?.[CREDITS_AMOUNT];
+  const currencyAmount = data?.[CURRENCY_AMOUNT];
+  const creditTypePrecision = creditTypeData?.creditType?.precision;
```

Comment:
Shouldn't we memoize this as well? It is used as a dependency for creating `creditsAvailable` and also passed as prop to `ChooseCreditsForm`. Actually, `ChooseCreditsForm` should be wrap with `React.memo()` too to prevent unnecessary re-renders.

Reply:
we use useMemo to wrap `creditsAvailable`
but we can use React.memo for ChooseCreditsForm too
I was also thinking about potentially creating a BuyCreditsContext, instead of having to pass all those props to the child components, wdyt? But maybe can be refactored later.

Reply:
My thinking is that since `creditTypePrecision` is recreated on every render, it will cause `creditsAvailable` to recalculate each time. Memoizing `creditTypePrecision` could prevent that, as well as unnecessary re-renders of `ChooseCreditsForm `.

I like your idea about a BuyCreditsContext, it will help simplify things.

Reply:
but `creditTypePrecision` doesn't change on every render, the actual value will only change once we get the correspond query response

Reply:
I believe the issue is that the variable is recreated and it's reference changes when react re-executes the component - even if its value remains the same - and that is what can trigger a `creditsAvailable` recalculation and a `ChooseCreditsForm` re-render

Reply:
but `creditTypePrecision` is a copy using some react query data which is cached and stable across re-renders (unless the data changes) so I don't think we need useMemo in this particular case
if that was another type of variable, then this would be useful indeed.

Reply:
You are right about the data being cached and stable across re-renders. However, my concern is that if the component re-renders for any reason, `creditTypePrecision` will be recreated, even if the value hasn't changed, which may lead to unnecessary recalculations for dependent variables or components. If we repeat this pattern throughout the app it might impact performance. 

That said, I’ll go ahead and approve the PR so we can move forward. Perhaps we can chat about this another time?